test/promql/tests/test_notready_pods.yaml

```diff
@@ -0,0 +1,67 @@
+evaluation_interval: 1m
+
+rule_files:
+  - ../extracted-rules.yaml
+
+tests:
+  - interval: 1m
+    input_series:
+
+      # Pod is in the Pending state and scheduled, so it will be alerted.
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-1", phase="Pending"}'
+        values: '1x15'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-1", phase="Unknown"}'
+        values: '0x15'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-1", phase="Failed"}'
+        values: '0x15'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-1", phase="Running"}'
+        values: '0x15'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-1", phase="Succeeded"}'
+        values: '0x15'
+      - series: 'kube_pod_status_unschedulable{namespace="ns-1", pod="pod-1"}'
+        values: '0x15'
+
+      # Pod is in the Unknown state and unscheduled(unscheduled once in the 15 min interval), so it will not be alerted.
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-3", phase="Pending"}'
+        values: '0x15'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-3", phase="Unknown"}'
+        values: '1x15'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-3", phase="Failed"}'
+        values: '0x15'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-3", phase="Running"}'
+        values: '0x15'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-3", phase="Succeeded"}'
+        values: '0x15'
+      - series: 'kube_pod_status_unschedulable{namespace="ns-3", pod="pod-1"}'
+        values: '1x1 0x14'
+
+      # Pod is in the Running state and scheduled, so it will not be alerted.
+      - series: 'kube_pod_status_phase{pod="pod-2", namespace="ns-1", phase="Running"}'
```

Comment:
Your code is not actually detecting the `Running` phase so I think you're missing the part of testing it. You need to also include the values of the metrics that your expression is checking. i.e. include the lines for all the bad phases and give them value of `0` throughout.